--- /data1/anguse/yingze/gromacs-2020.2/CMakeLists.txt	2021-09-16 00:25:17.125620740 +0800
+++ CMakeLists.txt	2021-09-16 14:54:13.434223510 +0800
@@ -134,6 +134,26 @@
 #    (i.e., something that is exposed in installed headers).
 set(GMX_PUBLIC_LIBRARIES "")
 
+# DeepMD
+message(STATUS "Compling with DeepMD...")
+add_definitions(-w) # close warning
+include_directories(${GMX_DEEPMD_ROOT}/include)
+include_directories(${GMX_TENSORFLOW_ROOT}/include)
+link_directories(${GMX_DEEPMD_ROOT}/lib)
+link_directories(${GMX_TENSORFLOW_ROOT}/lib)
+# define high precision
+add_definitions("-D HIGH_PREC") # only high precision of deepmd-kit is supported now
+# ensure all dynamic libraries linked
+set (CMAKE_EXE_LINKER_FLAGS "-Wl,--no-as-needed")
+# add link libraries
+list (APPEND GMX_PUBLIC_LIBRARIES deepmd_op)
+list (APPEND GMX_PUBLIC_LIBRARIES deepmd_op_cuda)
+list (APPEND GMX_PUBLIC_LIBRARIES deepmd)
+list (APPEND GMX_PUBLIC_LIBRARIES deepmd_cc)
+list (APPEND GMX_PUBLIC_LIBRARIES tensorflow_cc)
+list (APPEND GMX_PUBLIC_LIBRARIES tensorflow_framework)
+# DeepMD
+
 ########################################################################
 # Check and warn if cache generated on a different host is being reused
 ########################################################################
